runtime/lua/vim/treesitter/query.lua

```diff
@@ -92,6 +92,15 @@ local function read_query_files(filenames)
   return table.concat(contents, '\n')
 end
 
+local function new_match_metadata()
```

Comment:
now that metatable doesn't depend on hidden state it can just be one single named metatable :]